Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx

```diff
@@ -26,12 +31,40 @@ const SubmissionDetails = () => {
     const { actions: { setPageTitle } } = usePageTitles();
     const { state: { user: { permissions: { canAccessAdministration } } } } = useAuth();
     const { getAdministrationRetestSubmissionInternalUrl } = useAppUrls();
+    const {
+        state: { contest },
+        actions: { loadContestByProblemId },
+    } = useContests();
+
+    const { getRegisterContestTypeUrl } = useAppUrls();
+
+    useEffect(() => {
+        if (isNil(currentSubmission)) {
+            return;
+        }
+
+        const { problem: { id } } = currentSubmission;
+
+        loadContestByProblemId(id);
+    }, [ currentSubmission, loadContestByProblemId ]);
 
     const submissionTitle = useMemo(
         () => `Submission №${currentSubmission?.id}`,
         [ currentSubmission?.id ],
     );
 
+    const canBeCompeted = useMemo(
+        () => contest?.canBeCompeted,
```

Comment:
Is this needed to be in a useMemo?

Reply:
If I try to get it like this: const { canBeCompeted } = contest; - it does not work, because I have to check first for nullability of the contest, so using useMemo was the easiest solution to have canBeCompeted as a constant.